test/bdd/steps/endpoints.js

```diff
@@ -377,6 +378,103 @@ Given(
     },
 );
 
+Given(/^([DC|DH|DV]+) gets the list of available datasets for trading$/, async function (viewer) {
+    this.logger.log(`${viewer} gets the list of available datasets for trading.`);
+    expect(viewer, 'Node type can only be DC, DH, DV.').to.be.oneOf(['DC', 'DH', 'DV']);
+
+    const host = this.state[viewer.toLowerCase()].state.node_rpc_url;
+
+    const availableResponse = await httpApiHelper.apiPermissionedDataAvailable(host);
+    expect(availableResponse[0], 'Should have keys called dataset, ot_objects, seller_erc_id, seller_node_id, timestamp')
+        .to.have.all.keys('dataset', 'ot_objects', 'seller_erc_id', 'seller_node_id', 'timestamp');
+
+    const { dataset, ot_objects, seller_node_id } = availableResponse[0];
+    expect(this.state.lastImport.data.dataset_id).to.be.equal(dataset.id);
+    this.state.availablePurchase = {
+        data_set_id: dataset.id,
+        seller_node_id,
+        ot_object_id: ot_objects[0],
+    };
+});
+
+Given(/^([DC|DH|DV]+) gets the price for the last imported dataset$/, async function (viewer) {
+    this.logger.log(`${viewer} gets the price for the last imported dataset.`);
+    expect(viewer, 'Node type can only be DC, DH, DV.').to.be.oneOf(['DC', 'DH', 'DV']);
+
+    const host = this.state[viewer.toLowerCase()].state.node_rpc_url;
+
+    const { handler_id } = await httpApiHelper.apiPermissionedDataGetPrice(host, this.state.availablePurchase);
+    await sleep.sleep(2000);
+    const response = await httpApiHelper.apiPermissionedDataGetPriceResult(host, handler_id);
+
+    expect(response, 'Should have keys called data and status').to.have.all.keys('data', 'status');
+    expect(response.status).to.be.equal('COMPLETED');
+});
+
+
+Given(/^([DC|DH|DV]+) initiates purchase for the last imported dataset and waits for confirmation$/, async function (viewer) {
+    this.logger.log(`${viewer} initiates purchase for the last imported dataset and waits for confirmation.`);
+    expect(viewer, 'Node type can only be DC, DH, DV.').to.be.oneOf(['DC', 'DH', 'DV']);
+
+    const host = this.state[viewer.toLowerCase()].state.node_rpc_url;
+
+    const { handler_id } = await httpApiHelper.apiPermissionedDataPurchase(host, this.state.availablePurchase);
+    this.state.lastPurchaseHandler = handler_id;
+
+    this.state.lastQueryNetworkId = {};
+    this.state[viewer.toLowerCase()].state.purchasedDatasets = {};
+    this.state[viewer.toLowerCase()].state.purchasedDatasets[this.state.availablePurchase.data_set_id] = {};
+
+    const source = this.state.dc;
+
+    const promise = new Promise((acc, reject) => {
+        source.once('purchase-confirmed', async (data) => {
+            const target = this.state[viewer.toLowerCase()];
+            if (target.state.identity === data.dv_identity) { acc(); } else { reject(); }
+        });
+    });
+
+    return promise;
+});
+
+Given(/^(DC|DV|DV2) waits for purchase to finish$/, { timeout: 1200000 }, async function (targetNode) {
```

Comment:
Let's not use a 20-minute timeout for a step that does not have a rejection option